apollo-federation/src/sources/connect/url_template.rs

```diff
@@ -1,66 +1,47 @@
 use std::fmt::Display;
-use std::ops::Range;
 use std::str::FromStr;
 
 use apollo_compiler::collections::IndexMap;
 use itertools::Itertools;
 use serde::Serialize;
-use serde_json_bytes::ByteString;
-use serde_json_bytes::Map;
 use serde_json_bytes::Value;
-use serde_json_bytes::Value as JSON;
 use url::Url;
 
-use crate::sources::connect::variable::Namespace;
-use crate::sources::connect::variable::VariableError;
-use crate::sources::connect::variable::VariableReference;
+use crate::sources::connect::string_template::Error;
+use crate::sources::connect::string_template::Expression;
+use crate::sources::connect::string_template::StringTemplate;
 
 /// A parser accepting URLTemplate syntax, which is useful both for
 /// generating new URL paths from provided variables and for extracting variable
 /// values from concrete URL paths.
+///
+// TODO: In the future, when we add RFC 6570 features, this could contain one big
```

Comment:
We also want to allow use of `$config` in `base`, so this may need to change for that work as well.